shell/platform/fuchsia/flutter/vulkan_surface.cc

```diff
@@ -227,56 +239,113 @@ bool VulkanSurface::CreateFences() {
 }
 
 bool VulkanSurface::AllocateDeviceMemory(sk_sp<GrDirectContext> context,
-                                         const SkISize& size,
-                                         zx::vmo& exported_vmo) {
+                                         const SkISize& size) {
   if (size.isEmpty()) {
     return false;
   }
 
-  VulkanImage vulkan_image;
-  if (!CreateVulkanImage(vulkan_provider_, size, &vulkan_image)) {
-    FML_DLOG(ERROR) << "Failed to create VkImage";
+  fuchsia::sysmem::AllocatorSyncPtr sysmem_allocator;
+  zx_status_t status = fdio_service_connect(
+      "/svc/fuchsia.sysmem.Allocator",
+      sysmem_allocator.NewRequest().TakeChannel().release());
```

Comment:
I don't think we should connect to sysmem each time we allocate an image. Can this connection live in the surface producer and be reused for all allocations?

Reply:
+1, it is inefficient to do this every time we allocate an image